pkg/deployment/gitOps/git/GitOpsHelper.go

```diff
@@ -94,7 +95,8 @@ func (impl *GitOpsHelper) Pull(repoRoot string) (err error) {
 		util.TriggerGitOpsMetrics("Pull", "GitService", start, err)
 	}()
 	ctx := git.BuildGitContext(context.Background()).WithCredentials(impl.Auth)
-	return impl.gitCommandManager.Pull(ctx, repoRoot)
+	err = impl.gitCommandManager.Pull(ctx, repoRoot)
```

Comment:
create a wrapper for pull and throw err from there and the exposeed func should be used for matrix expose and wrapper calling.